js/isir-module-2025-2026.js

```diff
@@ -743,120 +351,166 @@ export const valid_country_codes = {
   'YD': 1, // Yemen
   'ZM': 1, // Zambia
   'ZW': 1, // Zimbabwe
+};
+
+//****************************
+// ISIR field validator logic implementations
+//
+
+const _validate_expect = (sz_value, field) => (sz_value == field.expect) || (field.allow_blank ? sz_value == '' : false);
+
+function _check_date(sz) {
+    if ('' === sz) return;
```

Comment:
What do you think about switching to TypeScript to take care of some of this validation? Checks like `isNaN` can have unexpected results in JavaScript.